doc/guidelines/api.md

```diff
@@ -0,0 +1,75 @@
+# REST
+
+We exclusively use:
+
+- JSON or images as content.
+- GET to retrieve resources.
+- POST to submit new resources to the server.
+- PUT to fully update an existing resource (fails if it does not exist).
+- PATCH to update parts of an existing resource (fails if it does not exist).
+- DELETE to remove resources.
+
+Both PUT and PATCH should be implemented idempotent.
+I.e., doing the same API calls again must be a NOP.
+
+## Endpoints
+
+The endpoint paths use:
+
+- hierarchical structure
+- nouns instead of verbs
+- only plural (exception: config)
+- all endpoints need authorization (exception: config)
+- all paths below `/api`
+- layer-specific paths below `/layers/<name of layer>/` e.g. `/layers/plants/plantings`
```

Comment:
```suggestion
- layer-specific paths below `maps/<map_id>/layers/<name of layer>/` e.g. `/layers/plants/plantings`
```
We will need the map_id as part of every request to the layer.
Should this be encoded in the path or as part of the request body?